samples/python_interop/qiskit.ipynb

```diff
@@ -0,0 +1,482 @@
+        "    # which is not supported by the Q# backend.\n",
+        "    cr3 = ClassicalRegister(\n",
+        "        name=\"cr3\",\n",
+        "        bits=[\n",
+        "            cr1[0],\n",
+        "            cr2[0],\n",
+        "        ],\n",
+        "    )\n",
+        "    qc = QuantumCircuit(q, cr1, cr2, cr3)\n",
+        "\n",
+        "    backend = QSharpBackend(target_profile=TargetProfile.Base)\n",
+        "    backend.qir(qc)\n",
+        "except QasmError as ex:\n",
+        "    print(f\"Exception: {str(ex)}\")\n",
+        "    # Print the cause of the exception if it exists.\n",
+        "    # This will print the error message from Qiskit itself.\n",
+        "    if ex.__cause__:\n",
+        "        print(f\"Cause: {str(ex.__cause__)}\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "62e33a40",
+      "metadata": {},
+      "source": [
+        "#### QIR generation semantic errors\n",
+        "\n",
+        "When targetting harware by compiling to QIR there are additional restrictions which may cause compilation errors. Most common scenarios:\n",
+        "- Trying to generate QIR when the profile is set to `Unrestricted`. `Unrestricted` is only valid for simulation. Either `TargetProfile.Base` or `TargetProfile.Adaptive_RI` must be used.\n",
+        "- Not all bits in classical registers have been assigned to. Usually because there were no measurements, or extra registers were declared.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "ac63ad88",
+      "metadata": {},
+      "source": [
+        "Example, generating QIR with `Unrestricted`"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "10d7dcb0",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "from qsharp.interop import QasmError\n",
+        "\n",
+        "try:\n",
+        "    circuit = QuantumCircuit(1)\n",
+        "    circuit.x(0)\n",
+        "    circuit.measure_all()\n",
+        "    backend = QSharpBackend()\n",
+        "    print(backend.qir(circuit))\n",
+        "except ValueError as ex:\n",
+        "    print(ex)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "fa219de1",
+      "metadata": {},
+      "source": [
+        "To avoid this issue, set the `target_profile` argument either in the `QSharpBackend` creation or in the `backend.qir` call."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "3170569e",
+      "metadata": {},
+      "source": [
+        "When generating `QIR`, all output registers must be read into before generating QIR. Failure to do so results in a `QSharpError`.\n",
+        "\n",
+        "In this first example, we declare two output bits, but only measure into one. This causes an error because result values can only be a side effect of measurement, and cannot be used like classical variables when compiling for hardware."
```

Comment:
> In this first example

There is only one example that follows. More to add here?

Reply:
Same as above, must have missed it when I was reorganizing and reordering. Thanks.